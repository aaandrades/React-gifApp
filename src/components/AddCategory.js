import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setinputValue] = useState("");

  const handleInputChange = (e) => {
    setinputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2 && inputValue !== "Enter your gif") {
      setCategories((values) => [inputValue, ...values]);
      setinputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        value={inputValue}
        placeholder="Enter your gif"
        onChange={handleInputChange}
        className="input-container"
        spellCheck="false"
      />
      <button className="button-container" onClick={handleSubmit}>
        Search
      </button>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
