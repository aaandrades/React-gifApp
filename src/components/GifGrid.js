import React from "react";
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {

  const { data: images } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.prototype = { 
  category: PropTypes.string.isRequired,
}
