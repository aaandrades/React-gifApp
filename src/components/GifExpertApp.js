import React, { useState } from "react";
import { AddCategory } from "./AddCategory";
import { GifGrid } from "./GifGrid";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Death Note"]);

  // const handleAdd = () => {
  //   setCategories([...categories, "Death Note"]);
  // };

  return (
    <div>
      <h2> Gif Expert App </h2>
      <AddCategory setCategories={setCategories} />
      <p className="Paragraph">Search the gif that you want and press enter!</p>
      <hr />
      <ol>
        {categories.map((category) => (
          <GifGrid 
              key={category} 
              category = {category} />
        ))}
      </ol>
    </div>
  );
};
