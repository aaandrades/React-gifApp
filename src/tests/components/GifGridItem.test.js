import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Test for GifGrifItem Component", () => {
  const title = "This is a title";
  const url = "https://localhost:8180/something.png";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);
  console.log("WRAPPER: ",wrapper)

  test("should render <GifGrifItem/> correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should have a paragraph with the Title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("should have an url", () => {
    const img = wrapper.find("img");
    expect(img.props().src).toBe(url);
  });

  test("should have an alternative Title", () => {
    const img = wrapper.find("img");
    expect(img.props().alt).toBe(title);
  });

  test("should have animate classname", () => {
    const div = wrapper.find("div");
    const className = div.prop("className");
    expect(className.includes("animate__animated")).toBe(true);
  });
});
