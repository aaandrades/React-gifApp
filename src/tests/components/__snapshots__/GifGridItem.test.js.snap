// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for GifGrifItem Component should render <GifGrifItem/> correctly 1`] = `
<div
  className="card animate__animated animate__fadeInUp"
>
  <img
    alt="This is a title"
    src="https://localhost:8180/something.png"
  />
  <p>
    This is a title
  </p>
</div>
`;
